R/modspec.R

```diff
@@ -39,6 +39,68 @@ write_capture <- function(x) {
   paste0("_capture_[",i-1,"] = ", x[i], ";") 
 }
 
+#' This function adds ETA values to the capture list when the `@etas` 
+#' option is used in `$CAPTURE` .
+#' 
+#' @param x the model object
+#' @param env the mread parse environment
+#' 
+#' @details
+#' We look at env$capture_etas to see if there were any expression text passed
+#' through the `@etas` option. This text will be parsed and evaluated in the 
+#' model environment after adding `LAST` and `last` to represent the last
+#' ETA (or the [total] number of rows in `$OMEGA`).
+#' 
+#' An error is generated in case the expression can't be parsed and evaluated.
+#' 
+#' `@etas` must resolve to an integer-like object. Expecting most usage to be
+#' `1:last` which will be integer, but we want to support `c(1,2,5)` as well
+#' which will not be integer. 
+#'
+#' @return The model object, possibly updated.   
+#' 
+#' @noRd
+capture_etas <- function(x, env) {
+  if(!is.character(env[["capture_etas"]])) return(x)
+  last <- dim_matlist(omat(x))
+  if(last==0) return(x)
+  parse_env <- as.list(env$ENV)
+  parse_env$last <- parse_env$LAST <- last
+  for(eta_txt in env[["capture_etas"]]) {
+    etan <- try(eval(parse(text = eta_txt), envir = parse_env), silent = TRUE)
+    if(inherits(etan, "try-error")) {
+      msg <- c(
+        glue("could not parse this expression for `etas`: {eta_txt}."), 
+        x = etan
+      )
+      abort(msg)
+    }
+    resolves_int <- 
+      is.integer(etan) || 
+      (is.numeric(etan) && all(abs(etan - round(etan)) < 1e-08))
```

Comment:
Fine as is, though perhaps it'd be better to match the same default tolerance of `sqrt(.Machine$double.eps)` (normally 1.490116e-08) that's used by `all.equal.numeric` and `dplyr::near` and in the example section of `help(integer)`.

Reply:
Thanks. I didn't think about `all.equal` or `near()`. Will switch to one of those. I was looking at `is_integerish()` which is what I really want but wasn't sure about the maturity of that one.